code_annotations/contrib/sphinx/extensions/featuretoggles.py

```diff
@@ -90,14 +92,28 @@ def iter_nodes(self):
                         toggle["line_number"],
                     ),
                 ),
+                ids="source",
             )
             toggle_section += nodes.paragraph(
-                text=toggle.get(".. toggle_description:", "")
+                text=f'Desc: {toggle.get(".. toggle_description:", "NaN")}',
+                ids="description",
             )
             if toggle.get(".. toggle_warning:") not in (None, "None", "n/a", "N/A"):
                 toggle_section += nodes.warning(
-                    "", nodes.paragraph("", toggle[".. toggle_warning:"])
+                    "", nodes.paragraph("", toggle[".. toggle_warning:"]), ids="warning"
                 )
+            optional_attrs = [
+                "creation_date",
+                "target_removal_date",
+                "implementation",
+                "use_cases",
+            ]
+            for opt in optional_attrs:
+                if toggle.get(f".. toggle_{opt}:") not in (None, "None", "n/a", "N/A"):
+                    toggle_section += nodes.paragraph(
+                        text=f'{opt.title().replace("_"," ")}: {toggle[f".. toggle_{opt}:"]}',
+                        ids=opt,
```

Comment:
Should `ids` be a list instead of a string? @timmc-edx noticed that the screenshot in https://github.com/openedx/code-annotations/pull/83#issuecomment-1238265335 shows a node being created for each character in, e.g. `creation_date`). See also [this docutils source code comment](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/nodes.py#l464).

Reply:
Good catch, yes that is correct, I have just did that also added toggle/setting name so we can unique ids, below is a screeshot:
![image](https://user-images.githubusercontent.com/16361375/190898785-e0e6cbe2-7ba1-44a9-a26a-57ce4737aa17.png)

Reply:
@pshiu Can I take this to mean security is ok with this PR merging?

Reply:
Yes! Please feel free to merge, @rgraber. Thank you for sending this in.

Notes from research:

> nodes.paragraph [[docutils source](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/nodes.py#l1698)] is TextElement [[docutils source](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/nodes.py#l1110)], but its children still accepts Inline elements (most notably images).
> 
> The general mechanic seems to be that we are creating a docutils document tree, which is then translated to HTML [[PEP design doc](https://peps.python.org/pep-0258/)]
> 
> When that translation happens in the HTML writer, it doesn't look like nasty things like <script> is supported [[list of supported tags](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/writers/html5_polyglot/__init__.py#l306)], so by default weird HTML tags are cast to a <span> [[else clause](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/writers/html5_polyglot/__init__.py#l327)], which is exactly what the PR requestor tested & found.